### PR TITLE
ARROW-4235: [GLib] Use "column_builder" in GArrowRecordBatchBuilder

### PR DESCRIPTION
**Pull request by shiro615**

### CODE REVIEW

File: c_glib/arrow-glib/table-builder.cpp [hunk 1]
```diff
@@ -241,6 +241,9 @@ garrow_record_batch_builder_get_n_fields(GArrowRecordBatchBuilder *builder)
  *   the `i`-th field on success, %NULL on out of index.
  *
  * Since: 0.8.0
+ *
+ * Deprecated: 0.13.0:
+ *   Use garrow_record_batch_builder_get_column_builder() instead.
```
Comment (kou) [inline c_glib/arrow-glib/table-builder.cpp]: Could you call  `garrow_record_batch_builder_get_column_builder()` in this function?

File: c_glib/arrow-glib/table-builder.h [hunk 2]
```diff
@@ -46,8 +46,15 @@ void garrow_record_batch_builder_set_initial_capacity(GArrowRecordBatchBuilder *
 GArrowSchema *garrow_record_batch_builder_get_schema(GArrowRecordBatchBuilder *builder);
 
 gint garrow_record_batch_builder_get_n_fields(GArrowRecordBatchBuilder *builder);
```
Comment (kou) [inline c_glib/arrow-glib/table-builder.h]: Could you also create `garrow_record_batch_builder_get_n_columns()`?

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
Could you check my comments?

File: c_glib/arrow-glib/table-builder.cpp (hunk 1 above)
Comment (shiro615) [reply c_glib/arrow-glib/table-builder.cpp]: Thank you for pointing out. I've fixed it.

File: c_glib/arrow-glib/table-builder.h (hunk 2 above)
Comment (shiro615) [reply c_glib/arrow-glib/table-builder.h]: I've created `garrow_record_batch_builder_get_n_columns()`.

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**